src/state.rs

```diff
@@ -20,8 +20,16 @@ pub struct ProcessElderChangeState {
     pub change_elder: Option<ChangeElder>,
 }
 
+#[derive(Debug, PartialEq, Default, Clone)]
+pub struct ProcessSplitState {
+    pub is_active: bool,
```

Comment:
Why do we need these `is_active` flags instead of trying all the flows one by one until one handles the event?

Reply:
All flows are not always active, some events are handled both by the main flow and one of its sub-flows (effectively shadowing), depending on if it is active

Reply:
I wonder if we can simply this though, while still allowing some flows to not be active all the time. For example, a flow can check some internal condition and only proceed if it's true. In this particular case that condition could be `churn_needed == Some(ChurnNeeded::Split)`. Maybe this can be extended to all flows?

Reply:
Then my suggestion would be to modify the diagrams too to make the flows more efficient. Probably out of scope for this PR though.

Reply:
Yes, I battled a bit to get readable diagrams, and this seemed to be working out ok, but is in no way perfect. Could be quite a bit of work so I'm not sure the benefit would be worth it.

Reply:
The thing I'm bit worried about is that once the work on routing_model is done and we start implementing this for real in routing, how strictly are we expected to follow the diagrams? One one hand, following them as closely as possible gives as some confidence that the logic is correct because it's already tested in routing_model. On the other hand, it can sometimes force us to write code that is not necessarily the most efficient.